### DISCUSSION TITLE
Maybe a bug : define filters on a query based data provider

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#6509] [Ideas]
I use postgreSQL.
If I use a table as source for my data providers : no problem I can define a filter.
If I use a query as source for my data providers : nothing happens when I click on "define filters" and nothing else works anymore in the designer, I must reload the page in my browser.
Tested with 2 different queries.

**Comment by NicolasFabre81** [comment]
In fact the point is that previously in the same data provider I had a table with filters.
And now in the query I do not have the same fields as in the previous table. So that causes the bug.

The solution is either to delete and re-create the DP, or to put back the previous table, delete the filters, then switch from the table to the query.

**Reply by melohagan** [reply]
Hey @NicolasFabre81 I tried adding a filter to a MySQL Data Provider, and switching between tables and custom queries and wasn't able to find any issue myself.
If you can list out steps to reproduce feel free to raise a bug!
